### PR TITLE
🌱 Add a prefix to the stack trace printed after SetLogger timeout

### PR DESCRIPTION
**Pull request by vincepri**
Example:
```bash
[controller-runtime] log.SetLogger(...) was never called; logs will not be displayed. Detected at:
	>  goroutine 1 [running]:
	>  runtime/debug.Stack()
	>  	/usr/local/go-faketime/src/runtime/debug/stack.go:24 +0x65
	>  main.main()
	>  	/tmp/sandbox4160348257/prog.go:13 +0x2b
```

### CODE REVIEW

File: pkg/log/log.go [hunk 1]
```diff
@@ -56,7 +57,15 @@ func eventuallyFulfillRoot() {
 	}
 	if time.Since(rootLogCreated).Seconds() >= 30 {
 		if logFullfilled.CompareAndSwap(false, true) {
-			fmt.Fprintf(os.Stderr, "[controller-runtime] log.SetLogger(...) was never called, logs will not be displayed:\n%s", debug.Stack())
+			stack := debug.Stack()
+			stackLines := bytes.Count(stack, []byte{'\n'})
+			sep := []byte{'\n', '\t', '>', ' ', ' '}
+
+			fmt.Fprintf(os.Stderr,
+				"[controller-runtime] log.SetLogger(...) was never called, logs will not be displayed:%s%s", sep,
```
Comment (seh) [inline pkg/log/log.go:65]: Is there an extra "%s" in that format string? It looks like we're only supplying one value to interpolate.

File: pkg/log/log.go [hunk 2]
```diff
@@ -56,7 +57,15 @@ func eventuallyFulfillRoot() {
 	}
 	if time.Since(rootLogCreated).Seconds() >= 30 {
 		if logFullfilled.CompareAndSwap(false, true) {
-			fmt.Fprintf(os.Stderr, "[controller-runtime] log.SetLogger(...) was never called, logs will not be displayed:\n%s", debug.Stack())
+			stack := debug.Stack()
+			stackLines := bytes.Count(stack, []byte{'\n'})
+			sep := []byte{'\n', '\t', '>', ' ', ' '}
+
+			fmt.Fprintf(os.Stderr,
+				"[controller-runtime] log.SetLogger(...) was never called, logs will not be displayed:.\nDetected at%s%s", sep,
```
Comment (seh) [inline pkg/log/log.go:65]: Just to confirm, did you mean to omit the colon after "at" and before the first separator on the second line of the message?

File: pkg/log/log.go [hunk 3]
```diff
@@ -56,7 +57,15 @@ func eventuallyFulfillRoot() {
 	}
 	if time.Since(rootLogCreated).Seconds() >= 30 {
 		if logFullfilled.CompareAndSwap(false, true) {
-			fmt.Fprintf(os.Stderr, "[controller-runtime] log.SetLogger(...) was never called, logs will not be displayed:\n%s", debug.Stack())
+			stack := debug.Stack()
+			stackLines := bytes.Count(stack, []byte{'\n'})
+			sep := []byte{'\n', '\t', '>', ' ', ' '}
+
+			fmt.Fprintf(os.Stderr,
+				"[controller-runtime] log.SetLogger(...) was never called; logs will not be displayed:.\nDetected at:%s%s", sep,
```
Comment (seh) [inline pkg/log/log.go:65]: I suggest deleting this extra colon character.
```suggestion
				"[controller-runtime] log.SetLogger(...) was never called; logs will not be displayed.\nDetected at:%s%s", sep,
```

**Comment by sbueringer** [comment]
Entirely missed this one. @vincepri can you address this one: https://github.com/kubernetes-sigs/controller-runtime/pull/2357#discussion_r1207023222. I think then we can merge it

**Comment by vincepri** [comment]
@sbueringer done

**Comment by sbueringer** [comment]
Thank you very much. Looks good

/lgtm
/approve

**Pull request merged**